runtime/sema/type.go

```diff
@@ -188,7 +189,7 @@ type TypeIndexableType interface {
 	Type
 	isTypeIndexableType() bool
 	IsValidIndexingType(indexingType Type) bool
-	TypeIndexingElementType(indexingType Type) Type
+	TypeIndexingElementType(indexingType Type, astRange ast.Range) (Type, error)
```

Comment:
Use interface `ast.HasPosition` instead of concrete `ast.Range`, so the range does not have to be created unless needed.

```suggestion
	TypeIndexingElementType(indexingType Type, positioned ast.HasPosition) (Type, error)
```

Reply:
Problem with this is that this function doesn't have a `checker` argument, which is needed to create a range from a positioned value. Is it worth adding this additional argument for this improvement?

Reply:
Ah, I see. In that case maybe just make the parameter a function `func() ast.Range`